### DISCUSSION TITLE
Is `-Yretain-trees` still poised to be enabled by default in the future?

### DISCUSSION THREAD

**Discussion by neko-kai** [scala/scala3#16176] [Metaprogramming]
We're porting a DSL to Scala 3 that requires inspecting the tree surrounding the macro call - and so requires `-Yretain-trees` to work consistently. I've been wary of using `-Yretain-trees`, but ran into [this comment](https://github.com/lampepfl/dotty/issues/12965#issuecomment-886581188) by @nicolasstucki saying that `-Yretain-trees` might eventually be enabled by default.
Is that still planned to happen? Would it be ok to require `-Yretain-trees` from users for now until it's enabled by default, or better find another way / change the DSL so that it's not required, because -Yretain-trees by default may not happen?

**Comment by joan38** [comment]
I have the same question as any project using [avro4s](https://github.com/sksamuel/avro4s) needs this option enabled to provide default values to Avro schema
